### PR TITLE
PS-1858 support loading files and tables to file workspaces

### PR DESCRIPTION
**Pull request by pivnicek**
ok, I think this is a good enough place to stop, with this they'll be able to import what they like into their workspaces.

### CODE REVIEW

File: kbcstorage/workspaces.py [hunk 1]
```diff
@@ -143,3 +144,26 @@ def load_tables(self, workspace_id, table_mapping, preserve=None):
         url = '{}/{}/load'.format(self.base_url, workspace_id)
 
         return self._post(url, data=body)
+
+    # only supports abs workspace
+    # the workspace must be the object returned by the create method otherwise it won't have the connectionString
+    def load_files(self, workspace, file_mapping, preserve=None):
```
Comment (odinuv) [inline kbcstorage/workspaces.py:150]: missing docs

File: kbcstorage/workspaces.py [hunk 2]
```diff
@@ -143,3 +144,26 @@ def load_tables(self, workspace_id, table_mapping, preserve=None):
...
+        if (workspace['type'] != 'file' and workspace['connection']['backend'] != 'abs'):
```
Comment (odinuv) [inline kbcstorage/workspaces.py:151]: i find it weird that the load_tables takes workspace_id and load_files takes workspace (with some "arbitrary" structure) do we need this check at all?

File: kbcstorage/workspaces.py [hunk 3]
```diff
@@ -143,3 +144,26 @@ def load_tables(self, workspace_id, table_mapping, preserve=None):
...
+        file_list = files.list(tags=file_mapping['tags'])
+        inputs = []
+        for file in file_list:
+            inputs.append({
+                'dataFileId': file['id'],
+                'destination': file_mapping['destination']
+            })
+        body = {}
+        i = 0
+        for input in inputs:
```
Comment (odinuv) [inline kbcstorage/workspaces.py:163]: use _make_body ?

File: tests/functional/test_workspaces.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        cursor = conn.cursor()
+        cursor.execute('SELECT * FROM "destination_1"')
+        self.assertEqual(('ping', 'pong'), cursor.fetchone())
+        cursor.execute('SELECT * FROM "destination_2"')
+        self.assertEqual(('king', 'kong'), cursor.fetchone())
+        conn.close()
+
+    # test load files into an abs workspace
+    def test_load_files_from_workspace(self):
+        if (os.getenv('SKIP_ABS_TEST')):
```
Comment (odinuv) [inline tests/functional/test_workspaces.py:97]: should be part of mock tests (if needed at all)

**Review by odinuv — COMMENTED** [review]
do we have to add the snowflake dependency? like - we can load and unload from the workspace to check that the table was loaded there and it only needs storage api

File: kbcstorage/workspaces.py (hunk 2 above)
Comment (pivnicek) [reply kbcstorage/workspaces.py:151]: omg, no, that was left from dev, I'll switch it back to id, thanks!

File: tests/functional/test_workspaces.py (hunk 4 above)
Comment (pivnicek) [reply tests/functional/test_workspaces.py:98]: @odinuv not sure what you mean by this, that loading files to the file workspace shouldn't be functionally tested?

Comment (pivnicek) [reply tests/functional/test_workspaces.py:101]: ah, if you meant the invalid workspace test throwing the exception then that makes sense, yeah that could go in the mock tests

File: kbcstorage/workspaces.py [hunk 5]
```diff
@@ -143,3 +144,30 @@ def load_tables(self, workspace_id, table_mapping, preserve=None):
         url = '{}/{}/load'.format(self.base_url, workspace_id)
 
         return self._post(url, data=body)
+
+    # only supports abs workspace
```
Comment (odinuv) [inline kbcstorage/workspaces.py:148]: this comment should be part of the docblock

**Review by odinuv — APPROVED** [review]
lgtm

**Pull request merged**